commands/Fun/rps.js

```diff
@@ -1,33 +1,33 @@
-exports.run = async (client, msg, [choice, user]) => {
-    if (!user) { user = client.user.id; }  
-    var data = await client.funcs.userSearch(msg, {user: [null, user], name: this.help.name});
-    
-    if (data.valid === false) { return; }
-    if (data.user[1].id === msg.author.id) { msg.channel.send("Hey! You can't play rock, paper, scissors with yourself! Invite someone into the mix or play with me instead!"); }
-        
-    var types = ["rock", "paper", "scissors"];
-    var hand = types[Math.floor(Math.random() * (Math.floor(2) - Math.ceil(1) + 1)) + Math.ceil(1)];
+const { Command } = require("klasa");
+
+module.exports = class extends Command {
+    constructor(...args) {
+        super(...args, {
+            name: "rps",
+            runIn: ["text"],
+            aliases: ["rockpaperscissors"],
+            description: "Play Rock, Paper, Scissors",
+            usage: "<paper|scissors|rock> [user:usersearch]", usageDelim: " "
+        });
+    }
 
-    if ((choice === "rock" && hand === "scissors") || (choice === "paper" && hand === "rock") || (choice === "scissors" && hand === "paper")) { var result = `**${data.user[0].prefered} wins!**`; } 
-    else if ((choice === "rock" && hand === "paper") || (choice === "paper" && hand === "scissors") || (choice === "scissors" && hand === "rock")) { var result = `**${data.user[1].prefered} wins!**`; }
-    else { var result = "**Draw!**"; }
+    async run(msg, [choice, user]) {
+        if (!user) { user === this.client.user; }
+        if (user.id === msg.author.id) { return msg.channel.send(this.client.speech(msg, ["rps", "sameuser"])); }
+        
+        var types = ["rock", "paper", "scissors"];
+        var hand = types[Math.floor(Math.random() * (Math.floor(2) - Math.ceil(1) + 1)) + Math.ceil(1)];  
+                
+        if ((choice === "rock" && hand === "scissors") || (choice === "paper" && hand === "rock") || (choice === "scissors" && hand === "paper")) { var result = `${msg.author.username} wins`; } 
+        else if ((choice === "rock" && hand === "paper") || (choice === "paper" && hand === "scissors") || (choice === "scissors" && hand === "rock")) { var result = `${data.user.username} wins`; }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['data' is not defined.](https://app.codacy.com/manual/Butterstroke/MargarineBot/pullRequest?prid=4918293)